### PR TITLE
feat: add support for user defined Table View Functions

### LINKED ISSUE #699: Add support for user defined TVF
This FR is to support management of User Defined Table Valued Functions (TVFs) using the routines API in BigQuery.  

Additional information has been provided to implementors separately.

### PR DESCRIPTION
**Pull request by plamut**
Closes #699.

Ta-da :tada: 

**PR checklist:**
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: google/cloud/bigquery/routine/routine.py [hunk 1]
```diff
@@ -16,11 +16,50 @@
...
+
 from google.protobuf import json_format
 
 import google.cloud._helpers
 from google.cloud.bigquery import _helpers
 import google.cloud.bigquery_v2.types
+from google.cloud.bigquery_v2.types import StandardSqlField
+
+
+# Unlike related proto-plus classes from `bigquery_v2.types.standard_sql`, this class
```
Comment (shollyman) [inline google/cloud/bigquery/routine/routine.py:29]: Sounds like we need to force an internal regen to get this picked up? @tswast and I need to sort this out.

File: google/cloud/bigquery/routine/routine.py [hunk 2]
```diff
@@ -16,11 +16,50 @@
...
+    def __eq__(self, other: Any):
+        if not isinstance(other, StandardSQLTableType):
+            return NotImplemented
+        return self.columns == other.columns
+
+    __hash__ = None  # Python does this implicitly for us, but let's be explicit.
+
+
+class RoutineType:
+    """The fine-grained type of the routine.
```
Comment (plamut) [inline google/cloud/bigquery/routine/routine.py:54]: Note to self - after `StandardSQLTableType` is autogenerated, we should manually include the remaining new classes to the docs. This one is one of them.

File: google/cloud/bigquery/routine/routine.py (hunk 1 above)
Comment (plamut) [reply google/cloud/bigquery/routine/routine.py:29]: Done [here](https://github.com/googleapis/python-bigquery/pull/777), will replace the hand-written class with the generated one.

**Review by tseaver — COMMENTED** [review]
Change to use the generated type LGTM.

**Review by tswast — APPROVED** [review]
Awesome!

**Pull request merged**